src/pytest_bdd/gherkin_parser.py

```diff
@@ -56,264 +49,56 @@
-        )
-
-
-@dataclass
-class Rule:
-    id: str
-    location: Location
-    keyword: str
-    name: str
-    description: str
-    tags: list[Tag]
-    children: list[Child]
-
-    @classmethod
-    def from_dict(cls, data: dict[str, Any]) -> Self:
-        return cls(
-            id=data["id"],
-            location=Location.from_dict(data["location"]),
-            keyword=data["keyword"],
-            name=data["name"],
-            description=data["description"],
-            tags=[Tag.from_dict(tag) for tag in data["tags"]],
-            children=[Child.from_dict(child) for child in data["children"]],
-        )
-
-
-@dataclass
-class Background:
-    id: str
-    location: Location
-    keyword: str
-    name: str
-    description: str
-    steps: list[Step]
-
-    @classmethod
-    def from_dict(cls, data: dict[str, Any]) -> Self:
-        return cls(
-            id=data["id"],
-            location=Location.from_dict(data["location"]),
-            keyword=data["keyword"],
-            name=data["name"],
-            description=data["description"],
-            steps=[Step.from_dict(step) for step in data["steps"]],
-        )
-
-
-@dataclass
-class Child:
-    background: Background | None = None
-    rule: Rule | None = None
-    scenario: Scenario | None = None
-
-    @classmethod
-    def from_dict(cls, data: dict[str, Any]) -> Self:
-        return cls(
-            background=Background.from_dict(data["background"]) if data.get("background") else None,
-            rule=Rule.from_dict(data["rule"]) if data.get("rule") else None,
-            scenario=Scenario.from_dict(data["scenario"]) if data.get("scenario") else None,
-        )
-
-
-@dataclass
-class Feature:
-    location: Location
-    language: str
-    keyword: str
-    tags: list[Tag]
-    name: str
-    description: str
-    children: list[Child]
-
-    @classmethod
-    def from_dict(cls, data: dict[str, Any]) -> Self:
-        return cls(
-            location=Location.from_dict(data["location"]),
-            language=data["language"],
-            keyword=data["keyword"],
-            tags=[Tag.from_dict(tag) for tag in data["tags"]],
-            name=data["name"],
-            description=data["description"],
-            children=[Child.from_dict(child) for child in data["children"]],
-        )
-
-
-@dataclass
-class GherkinDocument:
-    feature: Feature
-    comments: list[Comment]
-
-    @classmethod
-    def from_dict(cls, data: Mapping[str, Any]) -> Self:
-        return cls(
-            feature=Feature.from_dict(data["feature"]),
-            comments=[Comment.from_dict(comment) for comment in data["comments"]],
-        )
-
-
-def _to_raw_string(normal_string: str) -> str:
-    return normal_string.replace("\\", "\\\\")
+def replace_datatable_values(document: GherkinDocument) -> None:
+    """Replace all cell values in DataTables within a GherkinDocument using _to_raw_string."""
```

Comment:
why is this necessary? Maybe it should be done by `gherkin-official`?

Reply:
also, I'd rather avoid trying to patch datatables here at all, and instead handle it in `src/pytest_bdd/scenario.py`, since it's one place there, but many places here